### DISCUSSION TITLE
Read the http response from the workfow.

### DISCUSSION THREAD

**Discussion by anushreenk** [elsa-workflows/elsa-core#1154] [Q&A]
Hi Team,
I am planning to implement one workflow where I am using Http request , I want to read the response of http request, but I am not able to read it,Can anyone help me witg stepwise explanation how can i read it.
![image](https://user-images.githubusercontent.com/10956362/122450089-3d110f80-cfc4-11eb-9341-a5a6d8eb2615.png)

**Comment by sfmskywalker** [comment]
Sure, take a look at this animation:

![send-http-request-response-demo](https://user-images.githubusercontent.com/938393/122562397-5016f480-d043-11eb-9893-869dce3c5181.gif)

**Reply by simmiblog** [reply]
Hi @sfmskywalker,
How can we save this http response in a variable and print it using WriteLine activity

**Reply by sfmskywalker** [reply]
If your **SendHttpRequest** activity is called e.g. `SendHttpRequest1` which invokes e.g. `https://jsonplaceholder.typicode.com/todos/3`, then you can store the resulting object in a variable called e.g. `MyVariable` using the following JS expression:

```javascript
activities.SendHttpRequest1.ResponseContent;
```

This value is stored as an ExpandoObject, which means you can access its fields using dot-notation. For example, you could write its `title` field like using the following JS expression:

```javascript
getVariable('MyVariable').title;
```

For your reference, the JSON returned from the sample URL mentioned above looks like this:

```json
{
  "userId": 1,
  "id": 3,
  "title": "fugiat veniam minus",
  "completed": false
}
```

**Reply by qiqiqiyaya** [reply]
in 2.9.2 version .
update `activities.SendHttpRequest1.ResponseContent;`   to `activities.SendHttpRequest1.ResponseContent();`   
![image](https://user-images.githubusercontent.com/37217632/205230132-6063a6a4-9a2c-4f57-86a0-415ffd43d111.png)

**Comment by Rathna314** [comment]
activities.SendHttpRequest1.ResponseContent(); give the complete json object. How should we access the values in that object. Elsa has been so disappointing.

**Reply by cuong2195** [reply]
Hi @sfmskywalker. How can I read the response of SendHttpRequest from C# code. I prefer coding

**Reply by Rathna314** [reply]
I think you can follow this Sample : https://github.com/elsa-workflows/elsa-core/blob/2.13.x/src/samples/aspnet/Elsa.Samples.SendHttp/SendHttpWorkflow.cs
